src/GitHub.Exports/Models/BranchModel.cs

```diff
@@ -32,7 +32,8 @@ public BranchModel(LibGit2Sharp.Branch branch, IRepositoryModel repo, IGitServic
             Extensions.Guard.ArgumentNotNull(repo, nameof(repo));
             Name = DisplayName = branch.FriendlyName;
 #pragma warning disable 0618 // TODO: Replace `Branch.Remote` with `Repository.Network.Remotes[branch.RemoteName]`.
-            Repository = branch.IsRemote ? new LocalRepositoryModel(branch.Remote.Url, gitService) : repo;
+            // NOTE: This method expects a localPath not a URL!
+            Repository = branch.IsRemote ? gitService.CreateLocalRepositoryModel(branch.Remote.Url) : repo;
```

Comment:
Should we also refactor `BranchModel` to make it a model here or in another PR?

Reply:
I'd prefer to attack `BranchModel` in another PR because it's largely independent of `LocalRepositoryModel`.